File: pset1/mario/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

// Stating function build
bool build(int height);


int main(void)
{
    // Obtain height from user between 1 and 8, inclusive.
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Input height into build function
    build(height);
}

// Build function will create two mirrored pyramids with a double space inbetween that has the height the same size as base

bool build(int height)
{
    int high = height;
    int base = height;

    do
    {
        // Start with height - 1 spaces per line, and decrease until zero
        for (int i = base - 1; i > 0; i--)
        {
            printf(" ");
        }
        // Start with 1 block and increase until height # of blocks per line
        for (int j = base - 1; j < high; j++)
        {
            printf("#");
        }
        // Double space gap between pyramids
        printf("  ");
        // Start with 1 block and increase until height # of blocks per line
        for (int k = base - 1; k < high; k++)
        {
            printf("#");
        }
        printf("\n");
        base--;
    }
    while (base > 0);

    return true;
}

File: pset6/sentimental-mario-more/mario.py
from cs50 import get_int


# Get height input from user until an integer between 1 & 8 is received
height = get_int("Height: ")

while height < 1 or height > 8:
    print("Please pick an integer between 1 and 8.")
    height = get_int("Height: ")

# Loop through a height amount of rows
for h in range(height):
    # Create base for range of blocks printed
    r = h + 1

    # Print correct amount of empty spaces
    print(" " * (height - r), end="")

    # Print correct amount of #
    print("#" * r, end="")

    # Print two blank spaces
    print("  ", end="")

    # Print rest of blocks
    print("#" * r)
